platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -705,7 +705,7 @@ std::exception_ptr OfflineDatabase::clearAmbientCache() try {
 
     resourceQuery.run();
 
-    vacuum();
+    if (autopack) vacuum();
```

Comment:
Optional: We could replace all the `vacuum()` calls with a protected `mayPack()` doing the `autopack == true` check and move the code inside `::vacuum` to `::pack`.

Reply:
We've places where vacuum should be called unconditionally, e.g. at database creation in order to enable incremental vacuum